test/integration/test_staging.py

```diff
@@ -0,0 +1,97 @@
+import os
+import pytest
+import subprocess
+
+from subprocess import TimeoutExpired
+
+
+def describe_create():
+  def it_reports_created():
+    subprocess.run(
+      'hokusai registry push --force',
+      shell=True,
+      text=True,
+      timeout=30
+    )
+    resp = subprocess.run(
+      'hokusai staging create',
+      capture_output=True,
+      shell=True,
+      text=True,
+      timeout=10
+    )
+    if resp.returncode != 0:
+      print(resp.stderr)
+    assert resp.returncode == 0
+    assert 'Created configmap hokusai-integration-test-environment' in resp.stdout
+    assert 'deployment.apps/hokusai-integration-test-web created' in resp.stdout
+    assert 'service/hokusai-integration-test-web created' in resp.stdout
+    assert 'Created Kubernetes environment' in resp.stdout
+
+def describe_status():
+  def it_shows_resources():
+    resp = subprocess.run(
+      'hokusai staging status',
+      capture_output=True,
+      shell=True,
+      text=True,
+      timeout=10
+    )
+    if resp.returncode != 0:
+      print(resp.stderr)
+    assert resp.returncode == 0
+    assert 'Resources' in resp.stdout
+    assert 'deployment.apps/hokusai-integration-test-web' in resp.stdout
+    assert 'service/hokusai-integration-test-web' in resp.stdout
+    assert 'Pods' in resp.stdout
+
+def describe_update():
+  def it_reports_no_changes():
+    resp = subprocess.run(
+      'hokusai staging update',
+      capture_output=True,
+      shell=True,
+      text=True,
+      timeout=10
+    )
+    if resp.returncode != 0:
+      print(resp.stderr)
+    assert resp.returncode == 0
+    assert 'deployment.apps/hokusai-integration-test-web unchanged' in resp.stdout
+    assert 'service/hokusai-integration-test-web unchanged' in resp.stdout
+    assert 'Updated Kubernetes environment' in resp.stdout
+
+def describe_refresh():
+  def it_times_out():
+    # expect timeout due to minikube lacking ECR image pull permission
```

Comment:
This is a little unorthodox... how long does this test take to run?

Reply:
yeah, initially i wanted the test to confirm that the deployment is successfully refreshed, but ran into trouble with ECR authentication. hokusai running in circleci container has access to circleci context with aws keys, so hokusai itself can authenticate to ECR. but those creds are not accessible by k8s running inside minikube, and k8s needs them for image pull. so we can configure minikube/k8s with the creds, not sure if that's good idea in regards to security, but i came to the conclusion that for the purpose of testing 'hokusai staging refresh' it seems good enough that the test sees `Refreshing...`, so letting the test time out (the timeout is b/c of minikube/k8s retrying/failing image pull)

Reply:
on second thought, seems best for test to confirm successful refresh.

Reply:
If the creds are already in the circle ci context what is the security issue around propagating them to minikube? My assumption is that it's local to the circle container as well, in which case I agree just running it with the creds and testing successful refresh would probably be preferable to waiting for a timeout. Alternatively, setting a very low timeout for the test and just checking that the command is issued (rather than testing the timeout itself) is acceptable I think.